File: u1-becoming-a-developer/d03-javascript/student/star-wars/artoo.js
var called = window.alert("(beep-boop whrrrrrr bleep) Welcome to the R2D2 calculator!");
while(true) {
  var userInput = prompt("Please enter which operator you would like to use ('+', '-', '*', '/', '^', '^1/2', 'sin', 'cos', 'tan', '!'). Enter 'q' to exit.");
  if (userInput === 'q'){
    break;
  } else if (userInput === '+' || userInput === '-' || userInput === '*' || userInput === '/' || userInput === '^') {
    var frstOper = parseInt(prompt("Enter you first operand:"),10);
    var secOper = parseInt(prompt("Enter your second operand:"),10);
      // if (userInput === "+"){
      //   var solution = frstOper + secOper;
      // } else if (userInput === "-"){
      //   var solution = frstOper - secOper;
      // } else if (userInput === "*") {
      //   var solution = frstOper * secOper;
      // } else if (userInput === "/") {
      //   var solution = frstOper / secOper;
      // }
      switch (userInput){
        case '+':
            var solution = frstOper + secOper;
            break;
        case '-':
            var solution = frstOper - secOper;
            break;
        case '*':
            var solution = frstOper * secOper;
            break;
        case '/':
            var solution = frstOper / secOper;
            break;
        case '^':
            var solution = Math.pow(frstOper,secOper);
            break;
      }

    window.alert("The solution to " + frstOper + " " + userInput + " " + secOper + " is " + solution + ".");
  } else if(userInput === '^1/2' || userInput === 'sin' || userInput === 'tan' || userInput === 'cos'){
    var Oper = parseInt(prompt("Enter operand:"),10);
      switch (userInput){
        case '^1/2':
            var solution = Math.sqrt(Oper);
            break;
        case 'sin':
            var solution = Math.sin(Oper);
            break;
        case 'cos':
            var solution = Math.cos(Oper);
            break;
        case 'tan':
            var solution = Math.tan(Oper);
            break;
      }
    window.alert("The solution to " + Oper + " " + userInput + " is " + solution + ".");
  } else if (userInput === '!'){
      var factorial = parseInt(prompt("Enter operand:"),10);
      var solution = 1;
      for (var i = factorial; i > 0; i--){
        var x = solution * i;
        solution = x;
      }
      window.alert("The solution to " + factorial + " " + userInput + " is " + solution + ".");
  } else {
    window.alert("whheeaaarrrrr! beee-ooohhh) You did not enter a viable operator!");
  }
}

File: u4-ruby-on-rails/d17-ruby-slippers/student/homework/starwars/artoo.rb
puts "(beep-boop whrrrrrr bleep) Welcome to the R2D2 calculator!"

while true do
  puts "Please enter which operator you would like to use ('+', '-', '*', '/', '^', '^1/2', 'sin', 'cos', 'tan', '!'). Enter 'q' to exit."
  user_input = gets.chomp
  if user_input == 'q'
    exit
  elsif user_input == '+' or user_input == '-' or user_input == '*' or user_input == '/' or user_input == '^'
    puts "Enter your first operand:"
    first_operand = gets.chomp.to_i
    puts "Enter your second operand:"
    second_operand = gets.chomp.to_i

    case user_input
    when '+'
      solution = first_operand + second_operand
    when '-'
      solution = first_operand - second_operand
    when '*'
      solution = first_operand * second_operand
    when '/'
      solution = first_operand / second_operand
    when '^'
      solution = first_operand ** second_operand
    end

    puts "The solution to #{first_operand} #{user_input} #{second_operand} is #{solution}."
  elsif user_input == '^1/2' or user_input == 'sin' or user_input == 'tan' or user_input == 'cos'
    puts "Enter operand:"
    operand = gets.chomp.to_i

    case user_input
    when '^1/2'
      solution = Math.sqrt(operand)
    when 'sin'
      solution = Math.sin(operand)
    when 'cos'
      solution = Math.cos(operand)
    when 'tan'
      solution = Math.tan(operand)
    end

    puts "The solution to #{operand} #{user_input} is #{solution}."

  elsif user_input == '!'
    puts "Enter operand:"
    factorial = gets.chomp.to_i
    solution = 1
    for x in 0..factorial
      new_number = solution * x
      solution = x
    end
    puts "The solution to #{factorial} #{user_input} is #{solution}."
  else
    puts "whheeaaarrrrr! beee-ooohhh) You did not enter a viable operator!"
  end
end
